extensions/deterministic-prism-did-generation-proposal

```diff
@@ -0,0 +1,196 @@
+
+This Master Key acts as the root for all subsequent key derivations used to construct a PRISM DID. Because the PRISM DID itself is computed as a hash of the canonicalized initial DID Document, the DID can be deterministically regenerated from the same Master Key and context-specific inputs.
+This inicial creation event would be immediately followed by a update event with all the information a user may want to add to get to a state DID Document
+
+Once the DID is resolved (via the standard PRISM method), the corresponding DID Document can be inspected. For each public key in the document, its associated derivation path can be reconstructed using a deterministic scheme (e.g., based on key usage and type). This allows for re-derivation of the corresponding private keys from the Master Key, enabling recovery or signing without any key material being persistently stored.
+
+In cases where keys are missing, deprecated, or rotated, new keys can be deterministically generated by incrementing the relevant path segments (e.g., key index), ensuring consistent and verifiable updates to the DID state.
+
+The only requirement is that the initial input — the mnemonic phrase — must be retained. All other material (master key, derived keys, DID document, DID string) can be regenerated as needed.
+
+## Key Properties of this approach
+
+- ✅ **Deterministic**: The same mnemonic will always produce the same set of DIDs.
+- ✅ **Stateless**: No need to persist keys or DIDs — all are regenerated from the mnemonic.
+  - After determining the PRISM DID, the current state can be resolved/recovered from the Cardano blockchain for convenience.
+- ✅ **Compatible**: Fully compliant with the PRISM DID method spec.
+  - This proposal is a subset of what the DID PRISM method supports in the initial creation event.
+- ✅ **Secure**: The DID is cryptographically tied to the mnemonic-derived seed.
+
+The remainder of this specification defines the deterministic derivation path, how to construct a DID Document from the derived keys, and how to produce the final PRISM DID identifier.
+
+## Derivation Path Format
+
+We use the following derivation path for deterministic DID generation:
+
+```
+m / purpose' / method' / did-index / key-usage / key-index
+```
+
+Note: The MASTER key (and also the VDR key) is always secp256k1, according to the specifications. So there is just one public key.
+The public key material can be encoded as `ECKeyData` or `CompressedECKeyData`. For the sake of being deterministic we propose to only use `CompressedECKeyData` format for the first Master Key.
+
+
+### Path Component Definitions
+
+| Path Segment | Description |
+|--------------|-------------|
+| `purpose'`   | Set to `29` (anagram for ID in Hex 0x1D) (Note in [CIP-1852] is `1852`). |
+| `method'`    | Unique constant for PRISM DID method `29` |
```

Comment:
If method 29 represents a did:prism.
@yshyn-iohk You mention we should use something similar for the keys in did:peer. What number should we use?